.github/hooks/pre-commit

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+#
+# Install this using:
+# ln -s ../../.github/hooks/pre-commit .git/hooks/pre-commit
```

Comment:
You can install this to automatically format all changed dart files before committing.